.github/workflows/push.yml

```diff
@@ -0,0 +1,61 @@
+on: push
+name: Build and test
+jobs:
+  check_codestyle:
+    name: Codestyle
+    runs-on: ubuntu-latest
+    if: '!github.event.deleted'
+    steps:
+    - uses: actions/checkout@master
+    - name: Composer install
+      uses: MilesChou/composer-action@master
+      with:
+        args: install
+    - name: Check codestyle
+      uses: docker://php:7.3-alpine
+      with:
+        entrypoint: vendor/bin/phpcs
+
+  static_code_analysis:
+    name: Static Code Analysis
+    runs-on: ubuntu-latest
+    if: '!github.event.deleted'
+    steps:
+    - uses: actions/checkout@master
+    - name: Composer install
+      uses: MilesChou/composer-action@master
+      with:
+        args: install
+    - name: Static code analysis
+      uses: docker://php:7.3-alpine
+      with:
+        entrypoint: vendor/bin/phpstan
+        args: analyse
+
+  unit_tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    if: '!github.event.deleted'
+    strategy:
+      matrix:
+        php-version: ["7.2", "7.3", "7.4"]
+        composer-update: ["", "--prefer-stable --prefer-lowest"]
+    steps:
+    - uses: actions/checkout@master
+    - name: Installing PHP
+      uses: shivammathur/setup-php@master
+      with:
+        php-version: ${{ matrix.php-version }}
+        # These extensions are required by PHP 7.4
+        extension-csv: mbstring, xmlwriter
+    - name: Composer update
+      run: composer update ${{ matrix.composer-update }}
+    - name: Unit tests
+      uses: docker://php:7.3-alpine
```

Comment:
We need to use the locally installed version and not a Docker image for this to work :smile:.